File: cpp/src/Account.cpp
#include "Account.h"
#include <stdexcept>

// Constructor initializes the account with an ID and initial balance.
Account::Account(const std::string& account_id, int balance)
    : account_id(account_id), balance(balance) {}

// Deposits a specified amount into the account.
// Returns the updated account balance.
int Account::deposit(int amount) {
    if (amount <= 0) {
        throw std::invalid_argument("Deposit amount must be positive.");
    }
    balance += amount;
    return balance;
}

// Withdraws a specified amount from the account.
// Returns the updated account balance.
int Account::withdraw(int amount) {
    if (amount <= 0) {
        throw std::invalid_argument("Withdrawal amount must be positive.");
    }
    if (amount > balance) {
        throw std::invalid_argument("Insufficient balance.");
    }
    balance -= amount;
    return balance;
}

// Retrieves the current balance of the account.
int Account::get_balance() const {
    return balance;
}

// Retrieves the account ID.
std::string Account::get_account_id() const {
    return account_id;
}

File: python/atm/account.py
class Account:
    """
    Class representing a bank account.
    Attributes:
        account_id (str): Unique identifier for the account.
        balance (int): Current balance in the account.
    """

    def __init__(self, account_id, balance=0):
        """
        Initializes an account with an ID and an optional initial balance.
        Args:
            account_id (str): The unique ID for the account.
            balance (int, optional): Initial balance. Defaults to 0.
        """
        self.account_id = account_id
        self.balance = balance

    def deposit(self, amount):
        """
        Deposits a specified amount into the account.
        Args:
            amount (int): The amount to deposit.
        Returns:
            int: Updated account balance.
        Raises:
            ValueError: If amount is non-positive.
        """
        if amount <= 0:
            raise ValueError("Deposit amount must be positive.")
        self.balance += amount
        return self.balance

    def withdraw(self, amount):
        """
        Withdraws a specified amount from the account.
        Args:
            amount (int): The amount to withdraw.
        Raises:
            ValueError: If amount exceeds balance or is non-positive.
        Returns:
            int: Updated account balance.
        """
        if amount <= 0:
            raise ValueError("Withdrawal amount must be positive.")
        if amount > self.balance:
            raise ValueError("Insufficient balance.")
        self.balance -= amount
        return self.balance

    def get_balance(self):
        """
        Retrieves the current account balance.
        Returns:
            int: Current balance.
        """
        return self.balance
